services/headless-lms/src/models/pages.rs

```diff
@@ -1027,3 +1169,180 @@ WHERE p.chapter_id = $1
 
     Ok(pages)
 }
+
+/**
+Returns search results for a phrase i.e. looks for matches where the words come up right after each other
+*/
+pub async fn get_page_search_results_for_phrase(
+    conn: &mut PgConnection,
+    course_id: Uuid,
+    page_search_request: &PageSearchRequest,
+) -> ModelResult<Vec<PageSearchResult>> {
+    let course = crate::models::courses::get_course(&mut *conn, course_id).await?;
+
+    // Last word of the search term needed so that the sql statement can change it to a prefix match.
+    // Allows the last word to not be fully typed.
+    let last_word = page_search_request
+        .query
+        .trim()
+        .split_ascii_whitespace()
+        .last()
+        // If not found we'll skip the replacement with a random word not found in the search corpus
+        .unwrap_or("randomuniquewordforskippingthereplacement332");
```

Comment:
If not found the query is empty, might be OK to just return an empty vec, or consider an empty query to be an error by the client?